mqtt/mqtt-broker/src/configuration.rs

```diff
@@ -1,16 +1,59 @@
-use std::convert::From;
+use std::convert::{From, TryFrom};
 use std::ops::Mul;
-use std::path::Path;
+use std::path::{Path, PathBuf};
 use std::str::FromStr;
 use std::time::Duration;
 
+use crate::{transport::TransportBuilder, Error, ErrorKind};
 use config::{Config, ConfigError, File, FileFormat};
+use failure::ResultExt;
 use lazy_static::lazy_static;
+use native_tls::Identity;
 use regex::Regex;
 use serde::{Deserialize, Deserializer};
+use tracing::info;
 
 pub const DEFAULTS: &str = include_str!("../config/default.json");
 
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum Transport {
+    Tcp {
+        address: String,
+    },
+    Tls {
+        address: String,
+        srv_cert_path: PathBuf,
+    },
+}
+
+impl TryFrom<Transport> for TransportBuilder<String> {
+    type Error = Error;
+
+    fn try_from(transport: Transport) -> Result<Self, Self::Error> {
+        match transport {
+            Transport::Tcp { address } => Ok(Self::Tcp(address)),
+            Transport::Tls {
+                address,
+                srv_cert_path,
+            } => {
+                load_identity(srv_cert_path.as_path()).map(|identity| Self::Tls(address, identity))
+            }
+        }
+    }
+}
+
+fn load_identity(path: &Path) -> Result<Identity, Error> {
+    info!("Loading identity from {:?}", path);
+    let cert_buffer = std::fs::read(&path).context(ErrorKind::LoadIdentity)?;
+
+    let cert_pwd = "";
```

Comment:
I think this variable don't bring too much value. Shouldn't we inline it instead?